src/Core/Utils/Attribs.hpp

```diff
@@ -111,32 +113,43 @@ class Attrib : public AttribBase
     using value_type = T;
     using Container  = VectorArray<T>;
 
-    explicit Attrib( const std::string& name ) : AttribBase( name ) {}
-
+    explicit Attrib( const std::string& name );
+    virtual ~Attrib();
     /// Resize the container (value_type must have a default ctor).
-    void resize( size_t s ) override { m_data.resize( s ); }
-
+    void resize( size_t s ) override;
     /// Read-write access to the attribute content.
-    inline Container& data() { return m_data; }
+    /// lock the content, when done call unlock()
+    inline Container& getDataWithLock();
+    const void* dataPtr() const override;
+    ///@{
+    /// setAttribData, attrib mustn't be locked (it's asserted).
+    void setData( const Container& data );
+    void setData( Container&& data );
+    ///@}
 
     /// Read-only acccess to the attribute content.
-    inline const Container& data() const { return m_data; }
+    inline const Container& data() const;
 
-    virtual ~Attrib() { m_data.clear(); }
-    size_t getSize() override { return m_data.size(); }
+    size_t getSize() const override;
 
+    size_t getElementSize() const override;
     /// \warning Does not work for dynamic and sparse Eigen matrices.
-    int getStride() override { return sizeof( value_type ); }
-
-    bool isFloat() const override { return std::is_same<float, T>::value; }
-    bool isVec2() const override { return std::is_same<Eigen::Matrix<Scalar, 2, 1>, T>::value; }
-    bool isVec3() const override { return std::is_same<Eigen::Matrix<Scalar, 3, 1>, T>::value; }
-    bool isVec4() const override { return std::is_same<Eigen::Matrix<Scalar, 4, 1>, T>::value; }
+    int getStride() const override;
```

Comment:
Well stride is the number of bytes between to cell of the data array.
Actually I don't remeber if I checked that, and do not now how sparse and dynamic matrices are implemented.
But I imagine there is a heap allocation, and since attribs are set to the GPU as plain C arrays, it will miss the heap allocated mem.

So getStride will return something, but it do not correspond to the stride in a "opengl" way of thinking.
Maybe it's better to say that on the GPU classes sides